### PR TITLE
tests,network: Use alpine guest OS

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Before this PR:
Some network conformance tests use Cirros VMs.

After this PR:
Use Alpine VMs instead.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
This PR is also a workaround for https://github.com/kubevirt/kubevirtci/issues/1133 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by AlonaKaplan** [comment]
> Alpine has no dhcp client thus these cmds removed.

alpine has a dhcp client. It is called - udhcpc

> Also, the VM uses masquerade binding meaning the guest IP address doesnt change.

The test is testing the the guest DHCP client is capable to communicate with the new DHCP server (it may be need if the DHCP lease is over or the iface is going down/up). There was a [bug](https://github.com/kubevirt/kubevirt/pull/4804) in the past that the new DHCP server dropped requests from the guest.

**Comment by ormergi** [comment]
> > Alpine has no dhcp client thus these cmds removed.
> 
> alpine has a dhcp client. It is called - udhcpc

Thanks, fixed.

> > Also, the VM uses masquerade binding meaning the guest IP address doesnt change.
> 
> The test is testing the the guest DHCP client is capable to communicate with the new DHCP server (it may be need if the DHCP lease is over or the iface is going down/up). There was a [bug](https://github.com/kubevirt/kubevirt/pull/4804) in the past that the new DHCP server dropped requests from the guest.

Thanks for the heads up, I didnt know there is history for this test.
How about mentioning it in the assertion / step [message](https://github.com/kubevirt/kubevirt/blob/main/tests/network/vmi_networking.go#L896) so the next dev will think of removing it? 
Maybe something along these lines "verify the guest DHCP client is able to communicate with the pod DHCP server"?

**Comment by AlonaKaplan** [comment]
> > > Alpine has no dhcp client thus these cmds removed.
> > 
> > 
> > alpine has a dhcp client. It is called - udhcpc
> 
> Thanks, fixed.
> 
> > > Also, the VM uses masquerade binding meaning the guest IP address doesnt change.
> > 
> > 
> > The test is testing the the guest DHCP client is capable to communicate with the new DHCP server (it may be need if the DHCP lease is over or the iface is going down/up). There was a [bug](https://github.com/kubevirt/kubevirt/pull/4804) in the past that the new DHCP server dropped requests from the guest.
> 
> Thanks for the heads up, I didnt know there is history for this test. How about mentioning it in the assertion / step [message](https://github.com/kubevirt/kubevirt/blob/main/tests/network/vmi_networking.go#L896) so the next dev will think of removing it? Maybe something along these lines "verify the guest DHCP client is able to communicate with the pod DHCP server"?

`By("Initiating DHCP client request after migration")` can be extended to include the info.
Maybe "Initiating DHCP client request after migration to verify communication with the new DHCP server"?

File: tests/network/vmi_networking.go [hunk 1]
```diff
@@ -895,8 +898,8 @@ var _ = SIGDescribe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:c
 
 				By("Initiating DHCP client request after migration")
 
-				Expect(console.RunCommand(vmi, "sudo cirros-dhcpc down eth0\n", time.Second*time.Duration(15))).To(Succeed(), "failed to release dhcp client")
-				Expect(console.RunCommand(vmi, "sudo cirros-dhcpc up eth0\n", time.Second*time.Duration(15))).To(Succeed(), "failed to run dhcp client")
+				Expect(console.RunCommand(vmi, "udhcpc -i eth0\n", time.Second*time.Duration(15))).To(Succeed(),
```
Comment (AlonaKaplan) [inline tests/network/vmi_networking.go:901]: I think the lease has to be released first.

**Pull request closed without merging**